scripts/cfos/make_dataset_data_tables.py

```diff
@@ -87,113 +58,145 @@ def generate_load_table_files(dataset_tables_dict):
-        dataset_tables_dict[table] = table_df
-
-    print(f"Success: Dataset has been partitioned into {len(dataset_table_names)} tables: {dataset_table_names}")
-    return dataset_tables_dict
-
-
-def validate_dataset(dataset):
-    """Validate the full dataset metadata with Schema validator."""
-
-    # capture errors from the validation against the pre-defined validations per column
-    errors = schema_validator.validate(dataset, columns=schema_validator.get_column_names())
-    # get indices of the failed rows
-    errors_index_rows = [e.row for e in errors]
+def validate_dataset(dataset, schema_dict, field_dict):
+    validated_dataset = {}
+
+    for table in list(schema_dict.keys()):
+        if table not in list(dataset.keys()):
+            print(f"Failed: Missing required table {table}.")
+            print(dataset.keys())
+            exit()
+
+        errors = dynamically_validate_df(
+            field_dict=field_dict, 
+            fields_to_validate_list=schema_dict[table], 
+            data_df=dataset[table]
+            )
+        # get indices of the failed rows
+        errors_index_rows = [e.row for e in errors]
+
+        # if any errors, print error message and exit
+        if errors_index_rows:
+            print(f"Failed: Dataset validation errors found in table {table}. Please retry after correcting errors listed in validation_errors.csv.")
+            pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
+            exit()
+
+        # TODO: determine how to handle drops of failed rows/columns to get cleaned data
+        validated_dataset[table] = dataset[table]
+        print(f"Success: Table {table} has been validated.")
+    
+    print(f"Success: Dataset has been validated.")
+    return validated_dataset
 
-    # if any errors, print error message and exit
-    if errors_index_rows:
-        print(f"Failed: Dataset validation errors found. Please retry after correcting errors listed in validation_errors.csv.")
-        pd.DataFrame({'validation_error':errors}).to_csv('validation_errors.csv')
-        exit()
 
-    # TODO: determine how to handle drops of failed rows/columns to get cleaned data
-    # errors_index_rows = [e.row for e in errors]
-    # data_clean = dataset_metadata_df.drop(index=errors_index_rows, axis=1)
-    # data_clean.to_csv('clean_data.csv')
-    print(f"Success: Dataset has been validated.")
-    return dataset
+def col_is_in_defined_fields(col_name, field_dict):
+    list_of_fields = list(field_dict.keys())
+    if col_name in list_of_fields:
+        return True
 
+    return False
 
-def load_excel_input(excel, expected_dataset_cols):
-    """Read excel file input into a dataframe and validate for required columns."""
 
-    try:
-        dataset_metadata_df = pd.read_excel(excel, sheet_name="Sheet1", skiprows=2, usecols=expected_dataset_cols, index_col=None)
+def load_excel_input(excel, allowed_dataset_cols):
+    """
```

Comment:
Added: Returns: dictionary of data frames where each sheet is a separate dataframe in the dictionary

Reply:
ah thanks! I think it would be helpful to add a comment as well after line 122 with this info, which was not obvious to me
When `sheet_name=None` is specified, all sheets are returned, as a {sheet_name:dataframe} dictionary.

Reply:
What does adding the comment add that the docstring doesn't? Calling out that sheet_name=None is what causes this? Would I just be able to add that to the existing docstring instead of adding another comment elsewhere?

Reply:
yeah, exactly, that it's `sheet_name=None` that does this. and yeah I was suggesting adding it to your docstring after your other explanations of the parameters you pass to read_excel.

Reply:
added